### PR TITLE
Update Beats to the current Sarama fork

### PR DESCRIPTION
**Pull request by khushijain21**
This PR updates Beats to the current version of [Elastic's Sarama fork](https://github.com/elastic/sarama). It upgrades sarama version to 1.43.3. 

The import path has been changed from `github.com/Shopify/sarama` to `github.com/elastic/sarama` due to change in ownership.

### CODE REVIEW

**Comment by shmsr** [comment]
qq: Why don't we directly point to the fork i.e., github.com/elastic/sarama@beats-fork? Asking this because we have faced some problem with this before.

Read the PR description: https://github.com/elastic/beats/pull/39840#issue-2342416427. As it was breaking for the apm team, upon investigation, I found the problem and that's why we moved to using the fork directly. If required, I can also send the related Slack conversation in Slack.

So basically what'd happen is anyone using beats as a dependency, the replace directive won't work for them.

**Comment by khushijain21** [comment]
I think we can. @faec could you please confirm? Any particular reason we don't directly use the fork

**Comment by faec** [comment]
> could you please confirm? Any particular reason we don't directly use the fork

I don't see a reason we _can't_, but the reason we don't is probably that we would prefer to revert to upstream someday. Though there admittedly has not been much movement on that, so I don't think that's an overriding concern if it simplifies dependency handling overall.

File: go.mod [hunk 1]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/Microsoft/go-winio v0.6.2
 	github.com/PaesslerAG/gval v1.2.2
 	github.com/PaesslerAG/jsonpath v0.1.1
-	github.com/Shopify/sarama v1.27.0
+	github.com/Shopify/sarama v1.27.0 // indirect
```
[the remark this answers is not in the document]
Comment (mauri870) [reply go.mod:21]: The output of `go mod why github.com/Shopify/sarama` says:

```txt
# github.com/Shopify/sarama
github.com/elastic/beats/v7/metricbeat/module/kafka/consumergroup
github.com/elastic/beats/v7/metricbeat/module/kafka/consumergroup.test
github.com/bsm/sarama-cluster
github.com/Shopify/sarama
```

The `sarama-cluster` package is only used in a [single  integration test](https://github.com/elastic/beats/blob/2fec2d9/metricbeat/module/kafka/consumergroup/consumergroup_integration_test.go#L96-L109).

**Comment by pierrehilbert** [comment]
@lalit-satapathy could we have someone from your team reviewing here please?

File: testing/environments/docker/kafka/Dockerfile [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+ENV KAFKA_HOME=/kafka
 # Controls the hostname advertised within the Docker network, should generally match the container
 # name in a docker-compose file.
-ENV KAFKA_ADVERTISED_HOST kafka
+ENV KAFKA_ADVERTISED_HOST=kafka
 
 ENV KAFKA_LOGS_DIR="/kafka-logs"
-ENV KAFKA_VERSION 2.2.2
-ENV _JAVA_OPTIONS "-Djava.net.preferIPv4Stack=true"
+ENV KAFKA_VERSION=3.5.1
```
Comment (khushijain21) [inline testing/environments/docker/kafka/Dockerfile:9]: upgraded kafka version to recommended version as listed here https://docs.aws.amazon.com/msk/latest/developerguide/supported-kafka-versions.html

File: libbeat/common/kafka/version.go [hunk 3]
```diff
@@ -57,7 +57,16 @@ var (
 		"2.4": sarama.V2_4_0_0,
 		"2.5": sarama.V2_5_0_0,
 		"2.6": sarama.V2_6_0_0,
+		"2.7": sarama.V2_7_0_0,
```
Comment (khushijain21) [inline libbeat/common/kafka/version.go:60]: supported versions were referenced from here https://docs.aws.amazon.com/msk/latest/developerguide/supported-kafka-versions.html

Comment (mauri870) [reply libbeat/common/kafka/version.go:60]: Maybe we should add a comment referencing this link if we plan on following the same supported versions. When the time comes to revisit this list one can quickly get this list up to date when necessary. Could be right above `truncatedKafkaVersions` definition.

File: metricbeat/module/jolokia/_meta/Dockerfile [hunk 4]
```diff
@@ -6,7 +6,8 @@ ENV TC apache-tomcat-${TOMCAT_VERSION}
 ARG JOLOKIA_VERSION
 
 RUN apk update && \
-    apk add curl openssl ca-certificates bash
+    apk upgrade --no-cache && \
```
Comment (khushijain21) [inline metricbeat/module/jolokia/_meta/Dockerfile:9]: It uses an old version of `curl` which fails when you build locally. So this explicit upgrade was required

**Comment by khushijain21** [comment]
I think we should also update the kafka application version in our integration tests. I plan to do this in a separate PR.

**Review by rdner — APPROVED** [review]

**Review by mauri870 — APPROVED** [review]
LGTM, thanks!

**Review by shmsr — APPROVED** [review]

**Review by pierrehilbert — APPROVED** [review]
